ksqldb-common/src/main/java/io/confluent/ksql/reactive/BufferedPublisher.java

```diff
@@ -90,10 +90,10 @@ public BufferedPublisher(final Context ctx, final int bufferMaxSize) {
    */
   public boolean accept(final T t) {
     checkContext();
-    if (complete) {
+    if (isComplete() || hasSentComplete()) {
```

Comment:
If hasSentComplete, isn't complete always true?

Reply:
Not necessarily. A user could call `close()` on the publisher, which immediately calls `subscriber.onComplete()` (if a subscriber is present) without marking the publisher as complete (or waiting for any buffered rows to be delivered, in the case of BufferedPublisher). 

Is this not the expected behavior?